### PR TITLE
[Feat] Migrate and add new NSA Bwd operator 

### PR DESCRIPTION
**Pull request by wch0810**
## Description

Add Native Sparse Attention Backward (NSA Backward) kernel for Ascend NPU, implemented in `example_tilelang_nsa_bwd.py` based on TileLang Developer mode (hybrid) — `pass_configs` five-True (combineCV + auto_cv_sync + auto_sync + memory_planning + tail_mask).

Two `@tilelang.jit` kernels:
1. `nsa_fwd` — Forward (online softmax + sparse K block selection, for e2e tests)
2. `nsa_bwd_single` — Single-kernel backward (5 GEMM + softmax + Delta + dS, on-chip fusion)

`nsa_bwd_single` directly follows bhsd flashattn_bwd pattern: `alloc_shared/alloc_fragment` (compiler-mapped L1/UB/L0C), `threads=1`, on-chip direct `T.copy` (L0C→UB / UB→L1 — eliminates 9 GM workspace buffers). 2-level Compensated GEMM preserved (`ds_delta` + `ds_delta2`).

Layered test suite (20 cases) covering L0 threshold / L1 functional (including DeepSeek-V3 typical config D=64/D=128, BS=64) / L2 exception (blocking) / Boundary special-value (non-blocking).

Tested on Ascend NPU, fp16, B=1, T=32, H=1, HQ=16, D=32, S=1, BS=32. Precision verified against PyTorch CPU golden (single-pass softmax + autograd.backward, mixed-tolerance dual-gate per `precision-standard.md` §4.1: float16 atol=2⁻¹⁴, rtol=2⁻⁹, max_abs_limit=1e-1, required_ratio=0.99, **20/20 PASS**).
## Golden Config (B=1, T=32, H=1, HQ=16, D=32, S=1, BS=32, fp16)

| Kernel                | Task Duration (us) | Block Dim | Mix Block Dim | Op Type |
| --------------------- | ------------------ | --------- | ------------- | ------- |
| nsa_bwd_single | 198.46 (mean)    | 1         | 1             | mix     |

## Precision 

| Level              | Cases | Result | matched_ratio       | max_abs                  |
| ------------------ | ----- | ------ | ------------------- | ------------------------ |
| L0 (blocking)      | 1     | ✅ PASS | 0.9929 ~ 1.0000     | 1.953e-03                |
| L1 (blocking)      | 8     | ✅ PASS | 0.9929 ~ 1.0000     | 0 ~ 1.953e-03            |
| L2 (blocking)      | 3     | ✅ PASS | — (correctly rejected) | —                     |

- L1 includes 2 DeepSeek-V3 NSA paper typical config cases: `typical_d64_s8_bs64` (D=64, S=1, BS=64) + `typical_d128_bs64` (D=128, S=1, BS=64)
- L2 (blocking): 3 cases — `l2_unsupported_dtype` (fp32 input → ValueError/TypeError), `l2_illegal_shape` (D mismatch → AssertionError), `l2_zero_seqlen` (T=0 → AssertionError)
- Boundary (non-blocking): 2 WARN cases — `boundary_inf` (inf input propagates to dV/dK, expected), `boundary_param_s2` (S=2 exceeds NS=1 constraint, documented limitation)
- Thresholds (float16): atol=2⁻¹⁴ (6.10e-5), rtol=2⁻⁹ (1.95e-3), max_abs_limit=1e-1, required_ratio=0.99
- Dual-gate AND: matched_ratio ≥ 0.99 **AND** max_abs ≤ 1e-1
- INF/NAN structural compare (§3.1): positions must match, not counted in numeric tolerance

### CODE REVIEW

**Review by fengz72 — DISMISSED** [review]

File: examples_experiment/deepseek_nsa/example_tilelang_nsa_bwd/example_tilelang_nsa_bwd.py [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+
+    Hybrid mode (Developer + AUTO_CV_COMBINE/SYNC + T.serial).
+    Each block handles 1 Q token x 1 V slice. Iterates over S selected K blocks.
+    Uses T.tile.exp (natural log domain); scale = 1/sqrt(D) (no log2(e)).
+    LSE fix: m_i is already scaled, so no additional sm_scale multiplication.
+    """
+    sm_scale = (1.0 / dim) ** 0.5
+    G = groups
+    S = selected_blocks
+    BS = block_size
```
Comment (ChaoyangJi) [inline examples_experiment/deepseek_nsa/example_tilelang_nsa_bwd/example_tilelang_nsa_bwd.py:159]: 校验block_size的packed-mask和切分约束

实现同时使用BS/2在两个vid间切分KW行，并使用BS/8分配packed mask,但入口没有检查block size%8==0`。非8的倍数会导致T.tile.compare的256B对齐检查失败，或在后端mask容量不足；当前错误只能在较晚的lowering阶段暴露。请在fwd/bwd的入口统一增加明确的block_size约束，并补充非法block_size测试。

Comment (wch0810) [reply examples_experiment/deepseek_nsa/example_tilelang_nsa_bwd/example_tilelang_nsa_bwd.py:159]: 已补充
